server/src/main/java/org/elasticsearch/index/codec/tsdb/es819/ES819TSDBDocValuesProducer.java

```diff
@@ -1638,6 +1638,7 @@ private SortedNumericDocValues getSortedNumeric(SortedNumericEntry entry, long m
         final RandomAccessInput addressesInput = data.randomAccessSlice(entry.addressesOffset, entry.addressesLength);
         final LongValues addresses = DirectMonotonicReader.getInstance(entry.addressesMeta, addressesInput, merging);
 
+        assert entry.sortedOrdinals == null : "encoded ordinal range supports only one value per document";
```

Comment:
Invoking `getValues(...)` will result into a NPE if `entry.sortedOrdinals != null`. 
Maybe we can return a singleton `SortedNumericDocValues` instance here, so that at least the the first value of every document is available?